controllers/categories/services/create-category.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Request, Response } from "express";
+import { Reference } from "@firebase/database-types";
+import { ResponseHandler } from "../../../services";
+// require nanoid cause of typescript wahala
+const nanoid = require("nanoid");
+
+export async function createCategory(
+  req: Request,
+  res: Response,
+  usersRef: Reference
+) {
+  try {
+    const userId = req.session.userId,
+      // retrieve user from db
+      categoryRef = usersRef.child(userId).child("categories"),
+      // request body should contain name, description, image link (user will upload to firestore from FE), and object of tweet IDs.
+      categoryId = nanoid(),
+      category = categoryRef.child(categoryId);
+    await category.set(
+      {
+        name: req.body.name,
+        description: req.body.description,
+        image: req.body.image,
```

Comment:
We can destructure this at the top. 
`const { name, description, image } = req.body`